source/lac/cuda_precondition.cu

```diff
@@ -0,0 +1,1844 @@
+    AssertCusparse(status);
+
+    status = cusparseXcsrsv2_analysis(cusparse_handle,
+                                      CUSPARSE_OPERATION_TRANSPOSE,
+                                      n_rows,
+                                      n_nonzero_elements,
+                                      descr_L,
+                                      P_val_dev.get(),
+                                      P_row_ptr_dev,
+                                      P_column_index_dev,
+                                      info_Lt,
+                                      policy_Lt,
+                                      buffer_dev.get());
+    AssertCusparse(status);
+
+    status = cusparseXcsrsv2_analysis(cusparse_handle,
+                                      CUSPARSE_OPERATION_NON_TRANSPOSE,
+                                      n_rows,
+                                      n_nonzero_elements,
+                                      descr_L,
+                                      P_val_dev.get(),
+                                      P_row_ptr_dev,
+                                      P_column_index_dev,
+                                      info_L,
+                                      policy_L,
+                                      buffer_dev.get());
+    AssertCusparse(status);
+
+    // step 5: M = L * L'
+    status = cusparseXcsric02(cusparse_handle,
+                              n_rows,
+                              n_nonzero_elements,
+                              descr_M,
+                              P_val_dev.get(),
+                              P_row_ptr_dev,
+                              P_column_index_dev,
+                              info_M,
+                              policy_M,
+                              buffer_dev.get());
+    AssertCusparse(status);
+
+    int numerical_zero;
+    status =
+      cusparseXcsric02_zeroPivot(cusparse_handle, info_M, &numerical_zero);
+    AssertCusparse(status);
+  }
+
+
+
+  template <typename Number>
+  void
+  PreconditionIC<Number>::vmult(
+    LinearAlgebra::CUDAWrappers::Vector<Number> &      dst,
+    const LinearAlgebra::CUDAWrappers::Vector<Number> &src) const
+  {
+    Assert(P_val_dev != nullptr, ExcNotInitialized());
+    Assert(P_row_ptr_dev != nullptr, ExcNotInitialized());
+    Assert(P_column_index_dev != nullptr, ExcNotInitialized());
+    AssertDimension(dst.size(), static_cast<unsigned int>(n_rows));
+    AssertDimension(src.size(), static_cast<unsigned int>(n_rows));
+    Assert(tmp_dev != nullptr, ExcInternalError());
+
+    const Number *const src_dev = src.get_values();
+    Number *const       dst_dev = dst.get_values();
+    // step 6: solve L*z = alpha*x
+    const Number     alpha = internal::NumberType<Number>::value(1.);
+    cusparseStatus_t status =
+      cusparseXcsrsv2_solve(cusparse_handle,
+                            CUSPARSE_OPERATION_NON_TRANSPOSE,
+                            n_rows,
+                            n_nonzero_elem
```

Comment:
Typo

Reply:
What do you mean?

Reply:
This should be at line 1648 I don't why it shows the wrong line